### ISSUE TITLE
Redeclaring global variables should trigger "no-redeclare", not "no-undef".

### ISSUE THREAD

**Issue by impinball** [eslint/eslint#4504]
Minimal repro case:

```js
/* eslint no-undef: 2, no-redeclare: 2 */
var toString = 1;
```

Expected:

```
/home/isiah/dev-projects/js/ttyped/tmp.js
  2:5  error  "toString" is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

Found:

```
path/to/file.js
  2:5  error  "toString" is read only  no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by michaelficarra** [comment]
Why not?

**Comment by ilyavolodin** [comment]
`toString` is reserved JavaScript keyword and you are trying to override it. While it might not be clear why `no-undef` is the rule that's reporting this issue, it's a valid issue never the less.

**Comment by nzakas** [comment]
To be more precise, `toString` exists globally due to `Object.prototype.toString`, so it should behave like any other global.

That being said, `no-undef` should not be making this check as it should be covered by `no-redeclare`. We should fix this for 2.0.0.

**Comment by impinball** [comment]
Should there be an option to exclude Object prototype members?

Also, there's a lot of other cases where no-undef is erroneously complaining when other rules should be.

**Comment by nzakas** [comment]
Can you explain the other cases?

**Comment by impinball** [comment]
I meant other cases like the one above. The rest seem functionally identical to the fact it should be reported by `no-redeclare`. My bad on the part of really bad phrasing.

**Comment by impinball** [comment]
I updated the initial post.

**Comment by mysticatea** [comment]
Yes, `no-redeclare: [2, {builtinGlobals: true}]` is catching `var toString = 0`.
And `no-native-reassign` is catching `toString = 0`.

So I think `no-undef` 's message can be removed.

**Comment by mysticatea** [comment]
At the same time, I want to set `{builtinGlobals: true}` of `no-redeclare` by default. Thought?

**Comment by nzakas** [comment]
Why?

**Comment by mysticatea** [comment]
Because now `no-undef` is catching it by default.

**Comment by nzakas** [comment]
Ah, so you're saying if we changed `no-redeclare`, we could keep the same behavior?

**Comment by mysticatea** [comment]
Exactly!

**Comment by nzakas** [comment]
I see. My thought was that we want to make as few breaking changes as possible to do this, so I'd prefer to keep no-redeclare as it is and just change no-undef.

**Comment by mysticatea** [comment]
Hmm, OK.
